### ISSUE TITLE
Client side exception when using RadioButtonGroup and Binder

### ISSUE THREAD

**Issue by peholmst** [vaadin/framework#11112]
- Vaadin Framework version: 8.5.1 and 8.5.0
- Browser version: Chrome 68.0.3440.106 (Official Build) (64-bit)

When using a `RadioButtonGroup` and a binder and dynamically creating and removing bindings for the button group, certain steps will cause a client side exception to be thrown:

```
com.vaadin.client.ApplicationConfiguration
SEVERE: (TypeError) : Cannot read property 'v' of null
com.google.gwt.core.client.JavaScriptException: (TypeError) : Cannot read property 'v' of null
	at Unknown.VYc(com.vaadin.DefaultWidgetSet-0.js)
	at Unknown.KYc(com.vaadin.DefaultWidgetSet-0.js)
	at Unknown.Ogd(com.vaadin.DefaultWidgetSet-0.js)
	at Unknown.ahd(com.vaadin.DefaultWidgetSet-0.js)
	at Unknown.X3b(com.vaadin.DefaultWidgetSet-0.js)
	at Unknown.iWb(com.vaadin.DefaultWidgetSet-0.js)
	at Unknown.B3b(com.vaadin.DefaultWidgetSet-0.js)
	at Unknown.r6d(com.vaadin.DefaultWidgetSet-0.js)
	at Unknown.t0d(com.vaadin.DefaultWidgetSet-0.js)
	at Unknown.Z7d(com.vaadin.DefaultWidgetSet-0.js)
	at Unknown.ice(com.vaadin.DefaultWidgetSet-0.js)
	at Unknown.VVb(com.vaadin.DefaultWidgetSet-0.js)
	at Unknown.gWb(com.vaadin.DefaultWidgetSet-0.js)
	at Unknown.x3b(com.vaadin.DefaultWidgetSet-0.js)
	at Unknown.Ei(com.vaadin.DefaultWidgetSet-0.js)
	at Unknown.wi(com.vaadin.DefaultWidgetSet-0.js)
	at Unknown.Ii(com.vaadin.DefaultWidgetSet-0.js)
	at Unknown.Ci(com.vaadin.DefaultWidgetSet-0.js)
	at Unknown.$h(com.vaadin.DefaultWidgetSet-0.js)
	at Unknown.bi(com.vaadin.DefaultWidgetSet-0.js)
	at Unknown.eval(com.vaadin.DefaultWidgetSet-0.js)
	at Unknown.d(com.vaadin.DefaultWidgetSet-0.js)
```
The following sample code can be used to trigger the exception. Please note that the exception does not occur on Vaadin 8.4.5, only on 8.5.0 and 8.5.1.
```
public class MyUI extends UI {

    private RadioButtonGroup<Option> radioButtons;
    private CheckBoxGroup<Option> checkBoxes;
    private CheckBox useRadioButtons;
    private Binder<Bean> binder;

    @Override
    protected void init(VaadinRequest vaadinRequest) {
        final VerticalLayout layout = new VerticalLayout();
        
        binder = new Binder<>(Bean.class);
        binder.setBean(new Bean());

        useRadioButtons = new CheckBox("Radio buttons");
        useRadioButtons.addValueChangeListener(event -> updateBinder());

        radioButtons = new RadioButtonGroup<>();
        radioButtons.setItems(Option.values());

        checkBoxes = new CheckBoxGroup<>();
        checkBoxes.setItems(Option.values());

        updateBinder();

        layout.addComponents(useRadioButtons, radioButtons, checkBoxes);

        Label bugInstructions = new Label("Trigger the bug: <ul>" +
                "<li>Check <b>FIRST</b>.</li>" +
                "<li>Check <b>Radio buttons</b>.</li>" +
                "<li>Uncheck <b>Radio buttons</b>.</li>" +
                "<li>Check <b>SECOND</b>.</li>" +
                "<li>Uncheck <b>FIRST</b>.</li>" +
                "<li>Check <b>Radio buttons</b>.</li>" +
                "<li>There should be an exception in the browser console.</li>", ContentMode.HTML);
        layout.addComponent(bugInstructions);

        setContent(layout);
    }

    private void updateBinder() {
        checkBoxes.setVisible(!useRadioButtons.getValue());
        radioButtons.setVisible(useRadioButtons.getValue());
        if (useRadioButtons.getValue()) {
            binder.removeBinding(checkBoxes);
            binder.forField(radioButtons).bind(Bean::getOption, Bean::setOption);
        } else {
            binder.removeBinding(radioButtons);
            binder.forField(checkBoxes).bind(Bean::getOptions, Bean::setOptions);
        }
    }

    @WebServlet(urlPatterns = "/*", name = "MyUIServlet", asyncSupported = true)
    @VaadinServletConfiguration(ui = MyUI.class, productionMode = false)
    public static class MyUIServlet extends VaadinServlet {
    }

    public enum Option {
        FIRST, SECOND, THIRD
    }

    public static class Bean {

        private Set<Option> options = Collections.emptySet();

        public Set<Option> getOptions() {
            return options;
        }

        public void setOptions(Set<Option> options) {
            this.options = options;
            printOptions();
        }

        public Option getOption() {
            if (options.size() > 0) {
                return options.iterator().next();
            } else {
                return null;
            }
        }

        public void setOption(Option option) {
            setOptions(Collections.singleton(option));
        }

        private void printOptions() {
            System.out.println("Options: " + options);
        }
    }
}
```

**Comment by yoshie82** [comment]
Same problem here, but not a good example to reproduce...

Seems, that VRadioButtonGroup#buildOptions fails, because first item in 'items'-list is null
````java
    public void buildOptions(List<JsonObject> items) {
...
        // items.get(0) == null !
        while (i < items.size()) {
            updateItem(new RadioButton(groupId), items.get(i), true);
            i++;
        }
    }
````


````java
    private void updateItem(RadioButton button, JsonObject item,
            boolean requireInitialization) {
        if (requireInitialization) {
            getWidget().add(button);
            button.setStyleName("v-radiobutton");
            button.addStyleName(CLASSNAME_OPTION);
            button.addClickHandler(this);
        }

        String itemHtml = item
                .getString(ListingJsonConstants.JSONKEY_ITEM_VALUE); // << FAILS because item is null
        if (!isHtmlContentAllowed()) {
            itemHtml = WidgetUtil.escapeHTML(itemHtml);
        }
````

**Comment by rmcuenen** [comment]
I have the same issue with Vaadin 8.5.1. I use a dialog (Window) as editor and I also use a Binder to set the object properties. The RadioButtonGroup is fine at first initialization or when the value does not change between different objects. When, however, the value changes I get the same error when re-opening the window:

```java
public static enum SelectionType {
    ONE, TWO;
}

@Override
protected void init(VaadinRequest vaadinRequest) {
    final Window dialog = new Window("Dialog");
    final RadioButtonGroup<SelectionType> typeSelection = new RadioButtonGroup<>("Select Type", Arrays.asList(SelectionType.values()));
    final FormLayout layout = new FormLayout();
    layout.setWidthUndefined();
    layout.addComponent(typeSelection);
    dialog.setContent(layout);

    final VerticalLayout vLayout = new VerticalLayout();
    final Button button = new Button("Open");
    button.addClickListener(e -> {
        button.setEnabled(false);
        typeSelection.setSelectedItem(SelectionType.ONE);
        typeSelection.setSelectedItem(SelectionType.TWO);
        dialog.center();
        addWindow(dialog);
    });
    vLayout.addComponent(button);

    dialog.addCloseListener(e -> {
        button.setCaption("Re-open");
        button.setEnabled(true);
    });

    setContent(vLayout);
}
```

**Comment by TatuLund** [comment]
I think this should have been fixed by https://github.com/vaadin/framework/pull/11181 and is duplicate of #11143

**Comment by anasmi** [comment]
As Tatu noticed, this issue has been fixed in 8.6 version.

I've tried the code and there is no issue present in the console anymore (8.7.beta), but there is indeed one presented in 8.5.1.
As it seems, that issue is fixed, I will close this ticket, but , please, comment here if issue is still persisted under any circumstances or create a new ticket for us.

Thank you for contributions and help in finding bugs! :)
